File: SampleApp/Objective-C/SampleApp/Max/RewardedVideo/MaxRewardedVideoViewController.m
//
//  MaxRewardedVideoViewController.m
//  SampleApp
//
//  Copyright © 2022 fluct, inc. All rights reserved.
//

#import "MaxRewardedVideoViewController.h"
#import <AppLovinSDK/AppLovinSDK.h>
#import <FluctSDK/FluctSDK.h>

@interface MaxRewardedVideoViewController () <MARewardedAdDelegate>
@property (nonatomic, weak) IBOutlet UIButton *showButton;
@property (nonatomic, nullable) MARewardedAd *rewardedAd;
@end

// AdUnitIDを適切なものに変えてください
static NSString *const kAdUnitID = @"f681a5b35ac7822d";

@implementation MaxRewardedVideoViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (IBAction)didTouchUpLoadAdWithButton:(UIButton *)button {

    FSSRewardedVideoSetting *setting = FSSRewardedVideoSetting.defaultSetting;
    setting.debugMode = YES;
    setting.testMode = YES;

    [[FSSRewardedVideo sharedInstance] setSetting:setting];

    self.rewardedAd = [MARewardedAd sharedWithAdUnitIdentifier:kAdUnitID];
    self.rewardedAd.delegate = self;

    [self.rewardedAd loadAd];

    self.showButton.enabled = NO;
}

- (IBAction)didTouchUpShowAdWithButton:(UIButton *)button {
    if (self.rewardedAd.isReady) {
        [self.rewardedAd showAd];
    }
}

#pragma mark - MAAdDelegate Protocol

- (void)didLoadAd:(MAAd *)ad {
    NSLog(@"%s", __FUNCTION__);
    self.showButton.enabled = YES;
}

- (void)didFailToLoadAdForAdUnitIdentifier:(NSString *)adUnitIdentifier withError:(MAError *)error {
    NSLog(@"%s, %@", __FUNCTION__, error);
    self.showButton.enabled = NO;
}

- (void)didDisplayAd:(MAAd *)ad {
    NSLog(@"%s", __FUNCTION__);
}

- (void)didClickAd:(MAAd *)ad {
    NSLog(@"%s", __FUNCTION__);
}

- (void)didHideAd:(MAAd *)ad {
    NSLog(@"%s", __FUNCTION__);
    self.showButton.enabled = NO;
}

- (void)didFailToDisplayAd:(MAAd *)ad withError:(MAError *)error {
    NSLog(@"%s, %@", __FUNCTION__, error);
    self.showButton.enabled = NO;
}

#pragma mark - MARewardedAdDelegate Protocol

- (void)didStartRewardedVideoForAd:(MAAd *)ad {
    NSLog(@"%s", __FUNCTION__);
}

- (void)didCompleteRewardedVideoForAd:(MAAd *)ad {
    NSLog(@"%s", __FUNCTION__);
}

- (void)didRewardUserForAd:(MAAd *)ad withReward:(MAReward *)reward {
    NSLog(@"%s", __FUNCTION__);
}

@end

File: SampleApp/Swift/SampleApp/Max/RewardedVideo/MaxRewardedVideoViewController.swift
//
//  MaxRewardedVideoViewController.swift
//  SampleApp
//
//  Copyright © 2022 fluct, Inc. All rights reserved.
//

import UIKit
import FluctSDK
import AppLovinSDK

// AdUnitIDを適切なものに変えてください
private let adUnitID: String = "f681a5b35ac7822d"

class MaxRewardedVideoViewController: UIViewController {

    @IBOutlet weak var showButton: UIButton!

    private var rewardedAd: MARewardedAd!

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    @IBAction func didTouchUpLoadAd(button: UIButton) {
        // Fluct Setting
        let setting = FSSRewardedVideoSetting.default
        setting.isTestMode = true
        setting.isDebugMode = true

        FSSRewardedVideo.shared.setting = setting

        rewardedAd = MARewardedAd.shared(withAdUnitIdentifier: adUnitID)
        rewardedAd.delegate = self
        rewardedAd.load()

        showButton?.isEnabled = false
    }

    @IBAction func didTouchUpShowAd(button: UIButton) {
        if rewardedAd.isReady {
            rewardedAd.show()
        }
    }

}

extension MaxRewardedVideoViewController: MARewardedAdDelegate {

    // MARK: MAAdDelegate Protocol

    func didLoad(_ ad: MAAd) {
        print(#function)
        showButton?.isEnabled = true
    }

    func didFailToLoadAd(forAdUnitIdentifier adUnitIdentifier: String, withError error: MAError) {
        print(#function, error)
        showButton?.isEnabled = false
    }

    func didDisplay(_ ad: MAAd) {
        print(#function)
    }

    func didClick(_ ad: MAAd) {
        print(#function)
    }

    func didHide(_ ad: MAAd) {
        print(#function)
        showButton?.isEnabled = false
    }

    func didFail(toDisplay ad: MAAd, withError error: MAError) {
        print(#function, error)
        showButton?.isEnabled = false
    }

    // MARK: MARewardedAdDelegate Protocol

    func didStartRewardedVideo(for ad: MAAd) {
        print(#function)
    }

    func didCompleteRewardedVideo(for ad: MAAd) {
        print(#function)
    }

    func didRewardUser(for ad: MAAd, with reward: MAReward) {
        print(#function)
    }
}
